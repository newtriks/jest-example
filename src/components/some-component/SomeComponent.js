import React, { PropTypes } from 'react';
import CSSModules from 'react-css-modules';
// Styles
import styles from './styles.css';

const SomeComponent = ({ message }) => (
  <div>
    <p styleName="message">{message}</p>
  </div>
);

SomeComponent.propTypes = {
  message: PropTypes.string.isRequired
};

export default CSSModules(SomeComponent, styles);
