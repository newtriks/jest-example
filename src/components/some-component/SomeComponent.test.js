import React from 'react';
import SomeComponent from './SomeComponent';
import renderer from 'react-test-renderer';

it('renders SomeComponent as expected', () => {
  const msg = 'BOOM';
  const component = renderer.create(
    <SomeComponent message={msg} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
