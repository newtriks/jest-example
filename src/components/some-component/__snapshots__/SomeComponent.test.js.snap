exports[`test renders SomeComponent as expected 1`] = `
<div>
  <p
    className="message">
    BOOM
  </p>
</div>
`;
